x/merkledb/trieview.go

```diff
@@ -910,19 +927,20 @@ func (t *trieView) recordKeyChange(key path, after *node) error {
 		existing.after = after
 		return nil
 	}
+	t.changes.nodes[key] = &change[*node]{
+		after: after,
+	}
 
-	before, err := t.getParentTrie().getEditableNode(key)
-	if err != nil {
-		if err != database.ErrNotFound {
-			return err
+	if !newNode {
```

Comment:
Could we reverse this condition and reduce the indentation here? I think it'll make it more readable